### PR TITLE
[No Jira] Replace BpkRouterLink with home-grown component

### PR DESCRIPTION
**Pull request by shaundon**
`BpkRouterLink` is only used by the docs site, so we've decided to remove it as a component. It's a holdover from when the docs site was in the web repo. This will mean we can remove some dependencies from the web repo and simplify the codebase.

This PR pulls the component into the docs site and removes the dependency on it, so that it's no longer used anywhere.

### CODE REVIEW

**Review by olliecurtis — DISMISSED** [review]

File: docs/src/components/RouterLink/RouterLink.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  return (
+    <NavLink
+      className={classNames.join(' ')}
+      activeClassName={getClassName('bpkdocs-router-link--active')}
+      to={to}
+      {...rest}
+    >
+      {children}
```
Comment (georgegillams) [inline docs/src/components/RouterLink/RouterLink.js:43]: Why doesn't this render a BpkLink instead of using custom styles?

**Review by georgegillams — CHANGES_REQUESTED** [review]

File: docs/src/components/RouterLink/RouterLink.js (hunk 1 above)
Comment (georgegillams) [reply docs/src/components/RouterLink/RouterLink.js:43]: https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/docs/api/Link.md#component-reactcomponent

Comment (georgegillams) [reply docs/src/components/RouterLink/RouterLink.js:43]: In this case, FancyLink would be BpkLink

Comment (shaundon) [reply docs/src/components/RouterLink/RouterLink.js:43]: It uses the `BpkLink` mixin behind the scenes. This is literally just copied from the existing component so I don't think it matters tbh

Comment (georgegillams) [reply docs/src/components/RouterLink/RouterLink.js:43]: Dogfooding is important though, and it'll save us changing this component if we change the behaviour of `BpkLink` in the future.

Comment (shaundon) [reply docs/src/components/RouterLink/RouterLink.js:43]: Dammit why are you always right 😂

I'll change it, thanks for the feedback!

**Review by georgegillams — APPROVED** [review]

**Pull request merged**